backend/app/services/splunk.py

```diff
@@ -110,6 +109,60 @@ async def _stream_results(self, oneshotsearch_results):
         for record in results.JSONResultsReader(oneshotsearch_results):
             yield record
 
+    async def filterPost(self, query, searchList=""):
+        """
+        Query data to construct filter from splunk server using splunk lib sdk
+
+        Args:
+            query (string): splunk query
+            OPTIONAL: searchList (string): additional query parameters for index
+        """
+
+        try:
+            # If additional search parameters are provided, include those in searchindex
+            searchindex = (
+                "search index={} {}".format(self.indice, searchList)
+                if searchList
+                else "search index={}".format(self.indice)
+            )
+            # cluster_artifacts.cpu_models{}.cpu
+            search_query = (
+                "search index={} {} | stats count AS total_records, values(cpu) AS cpu, values(node_name) AS nodeName, values(test_type) AS benchmark, values(ocp_version) AS ocpVersion, values(ocp_build) AS releaseStream".format(
+                    self.indice, searchList
+                )
+                if searchList
+                else "search index={} | stats count AS total_records, values(cpu) AS cpu, values(node_name) AS nodeName, values(test_type) AS benchmark, values(ocp_version) AS ocpVersion, values(ocp_build) AS releaseStream".format(
+                    self.indice
+                )
+            )
+
+            try:
+                # Run the job and retrieve results
+                job = self.service.jobs.create(
+                    search_query,
+                    exec_mode="blocking",
+                    earliest_time=query["earliest_time"],
+                    latest_time=query["latest_time"],
+                )
+                # Wait for the job to finish
+                while not job.is_done():
```

Comment:
Is a polling loop really common practice here? It doesn't seem like a great idea. If there's no way to *block* until a job is done, maybe there should be a short `wait` here?

Reply:
Removed that piece of code and added `await` in the `create` jobs line.